frontend/src/upload.js

```diff
@@ -23,21 +26,23 @@ let onChange = event => {
   fileSender.upload().then(info => {
     const url = `${window.location
       .origin}/download/${info.fileId}/#${info.secretKey}`;
+    link.innerHTML = url;
     localStorage.setItem(info.fileId, info.deleteToken);
-    link.innerText = url;
-    link.setAttribute('href', url);
-    let btn = document.createElement('button');
-    btn.innerText = 'Delete from server';
+    var del = document.createElement("td");
+    var btn = document.createElement("button");
+    btn.innerHTML = "x";
+    btn.style = "padding: 0; border: none; background: none; cursor: pointer"
     btn.addEventListener('click', () => {
       FileSender.delete(
         info.fileId,
         localStorage.getItem(info.fileId)
       ).then(() => {
-        document.getElementById('uploaded_files').removeChild(li);
+        document.getElementById('uploaded-files').removeChild(row);
```

Comment:
you're searching for the  'uploaded-files' element more than once, do you want to make it a variable?